pinot-common/src/main/java/org/apache/pinot/common/config/TableConfig.java

```diff
@@ -118,14 +118,24 @@ public static TableConfig fromJsonString(String jsonString)
   @Nonnull
   public static TableConfig fromJSONConfig(@Nonnull JsonNode jsonConfig)
       throws IOException {
+    // Mandatory fields
+    Preconditions.checkState(jsonConfig.has(TABLE_TYPE_KEY), "Table type is missing");
     TableType tableType = TableType.valueOf(jsonConfig.get(TABLE_TYPE_KEY).asText().toUpperCase());
+    Preconditions.checkState(jsonConfig.has(TABLE_NAME_KEY), "Table name is missing");
     String tableName = TableNameBuilder.forType(tableType).tableNameWithType(jsonConfig.get(TABLE_NAME_KEY).asText());
-
+    Preconditions
+        .checkState(jsonConfig.has(VALIDATION_CONFIG_KEY), "Mandatory config '%s' is missing", VALIDATION_CONFIG_KEY);
     SegmentsValidationAndRetentionConfig validationConfig =
         extractChildConfig(jsonConfig, VALIDATION_CONFIG_KEY, SegmentsValidationAndRetentionConfig.class);
+    Preconditions.checkState(jsonConfig.has(TENANT_CONFIG_KEY), "Mandatory config '%s' is missing", TENANT_CONFIG_KEY);
     TenantConfig tenantConfig = extractChildConfig(jsonConfig, TENANT_CONFIG_KEY, TenantConfig.class);
+    Preconditions
+        .checkState(jsonConfig.has(INDEXING_CONFIG_KEY), "Mandatory config '%s' is missing", INDEXING_CONFIG_KEY);
```

Comment:
Are INDEXING and CUSTOM configs really mandatory?

Reply:
Currently they are mandatory, and all the callers assume they are non-null.